src/pages/_app.tsx

```diff
@@ -11,10 +11,36 @@ import { Sidebar } from '../components/Sidebar/Sidebar';
 import HttpService from '../utils/HttpService';
 import { ROUTES } from '../constants/constants';
 import { useStore } from '../store/store';
+import { useRouter } from 'next/router';
 
-const init = async () => {
+const init = async pathname => {
   const cookbooks = await HttpService.get(ROUTES.COOKBOOKS);
   useStore.setState({ cookbooks });
+
+  const cookbookId = getCookBookIdIfExists(pathname);
+
+  if (cookbookId) {
+    const cookbook = await HttpService.get(ROUTES.COOKBOOK(cookbookId));
+    useStore.setState({ cookbook });
+  }
+
+  let guides = [];
+
+  cookbooks.forEach(async cookbook => {
+    const guidesInCookbook = await HttpService.get(ROUTES.GUIDES(cookbook._id));
+
+    guides = guides.concat(guidesInCookbook);
+
+    useStore.getState().updateGuides(guides);
```

Comment:
This seems like something that could be done within a fetch action within the state instead rather than the app file 
https://github.com/pmndrs/zustand#async-actions

Reply:
`forEach` and async probably dont work the way you think. In this case its probably fine since we dont care about it doing things in order but for future reference it's generally safer to use vanilla for loops with async

Reply:
+1 to @bartdebever

Reply:
also with state stuff its usually better to make a new array instead of using `.concat`. This might be fine with zustand but you could run into issues in the future so its better to do `guides = [...guidesInCookbook, ...guides]`

Reply:
concat returns a new array, so i think those would be the same?

i'll peep the rest tho

Reply:
Oh thats true, I would have to read more up on the differences. Also, Im pretty sure the cookbooks are already populated on the cookbook when it comes back from the API so you probably shouldnt have to do this at all.